src/components/Identity/visitorService/awaitVisitorOptIn.js

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { isObject } from "../../../utils";
+
+export default ({ logger }) => {
+  return new Promise(resolve => {
+    if (isObject(window.adobe) && isObject(window.adobe.optIn)) {
+      const optInOld = window.adobe.optIn;
+      logger.log(
+        "Delaying request while waiting for legacy opt in to let Visitor retrieve ECID from server."
+      );
+      optInOld.fetchPermissions(() => {
+        if (optInOld.isApproved([optInOld.Categories.ECID])) {
+          const message =
+            "Received legacy opt in approval to let Visitor retrieve ECID from server.";
+
+          logger.log(message);
+          resolve(message);
+        }
+      }, true);
+    } else {
+      resolve("Legacy opt in object does not exist.");
```

Comment:
Hey @wgamboa, I noticed we're resolving these promises with a string, but we don't do anything with that resolved value. Can you explain why we don't just resolve the promise without the string?

Reply:
Hey @Aaronius, I use it for the unit tests. I don't know how else to go about it.

Reply:
I just remembered you advising me to avoid strings in the code. Maybe a numerical value?

Reply:
I don't think a string or a number provides any value here. If you test:

1. The promise is rejected if `adobe.optIn` doesn't exist.
1. The promise is resolved if `adobe.optIn` does exist and the user approves.
1. The promise is rejected if `aodbe.optIn` does exist and the user disapproves. I think? Should it be rejected if the user disapproves? It doesn't seem to be resolved or rejected at this point.

...then I don't think there's any value in checking a resolution _value_.

Reply:
1. I don't know what Minimum version of Visitor we should support. Should it have optin? If that's not required, then I think this promise should be resolved with undefined.
3. Thanks for the catch. This should be rejected with undefined